tao_compiler/mlir/disc/transforms/lhlo_fusion.cc

```diff
@@ -475,13 +487,20 @@ struct DiscFusionPass : public DiscFusionPassBase<DiscFusionPass> {
       pipeline.emplace_back(
           makeNewPlacementAwareFusionStrategy(gpu_enabled_, "base"));
     } else if (fusion_strategy_ == "stitch") {
-      // Do some basic fusion first.
-      pipeline.emplace_back(
-          makeNewPlacementAwareFusionStrategy(gpu_enabled_, "stitch_base"));
-      pipeline.emplace_back(
-          makeNewPlacementAwareFusionStrategy(gpu_enabled_, "stitch"));
-      pipeline.emplace_back(
-          makeNewPlacementAwareFusionStrategy(gpu_enabled_, "base"));
+      if (gpu_enabled_) {
+        pipeline.emplace_back(
```

Comment:
Run 'base' fusion strategy before 'stitch' fusion strategy may lose some fusion opportunities.

Say base fusion strategy supports op type: `{A, B, C}`, while 'stitch' fusion strategy supports op type: `{B, C}`

After the 'base' fusion, some fusion pattern may contains op A, making the following stitch fusion fail.

To minimize to impact of the above problem, stitch cpu implementation uses a three stage fusion pipeline:
- stage 0, run base fusion strategy, while using  the supported op list same as the stitch fusion strategy.
- stage 1, stitch fusion
- stage 2, normal base fusion strategy.

Reply:
TODO.